cudamapper/src/main.cu

```diff
@@ -311,9 +331,7 @@ int main(int argc, char* argv[])
                     //Need to perform this check again for thread-safety
                     if (range_idx < get_size<int>(query_target_ranges))
                     {
-                        auto overlap_future = compute_overlaps(query_target_ranges[range_idx], device_id);
-                        std::lock_guard<std::mutex> lck(overlap_futures_mtx);
-                        overlap_futures.push_back(overlap_future);
+                        compute_overlaps(query_target_ranges[range_idx], device_id);
```

Comment:
No. Each invocation of `compute_overlaps` has its own local copy of an overalps vector (i.e `std::vector<Overlap>`. the only global mutex which compute_overlaps has access to is `std::mutex overlaps_writer_mtx`. This is there to prevent spamming of stdout. I have added a comment to clarify.

Reply:
Ok